src/reducers/existingThoughtChange.js

```diff
@@ -68,13 +69,34 @@ export const existingThoughtChange = (state, { oldValue, newValue, context, show
       .removing if the old path is already in the object
       .Checking for all the descendants and updating their path too
       .limiting number of recenlty edited thoughts
-      .Adding new thought to the array */
+      .Adding new thought to the array
+      .merging the paths with common majority subcontext
+      .concating recently edited thought path only if no merge has happened
+
+  */
+
+  let isMerged = false // eslint-disable-line fp/no-let
+
+  /* if we dont use isMerged variable we have have to iterate the array one more time to find if any merge has happened */
+
+  // const isMerged = recentlyEditedUdpate.some((recentlyEditedThought) => {
+  //   return checkIfPathShareSubcontext(recentlyEditedThought.path, newPath).isSubcontext
+  // })
 
   const recentlyEdited = reverse(sortBy([...state.recentlyEdited], 'lastUpdated'))
     .filter(recentlyEditedThought => !equalPath(recentlyEditedThought.path, oldPath))
     .map(recentlyEditedThought => subsetThoughts(recentlyEditedThought.path, oldPath) ? Object.assign({}, recentlyEditedThought, { path: newPath.concat(recentlyEditedThought.path.slice(newPath.length)) }) : recentlyEditedThought)
     .slice(0, RECENTLY_EDITED_THOUGHTS_LIMIT)
-    .concat(({ path: newPath, lastUpdated: timestamp() }))
+    .map((recentlyEditedThought) => {
+      // checking for availability of majoirty subcontext between two rankedThoughts[]
+      const subcontextData = checkIfPathShareSubcontext(recentlyEditedThought.path, newPath)
+      // this variable sets to true when there is atleast one recently edited thought it can merge to
+      if (subcontextData.isSubcontext) isMerged = true // eslint-disable-line fp/no-mutating-methods
+
+      // here returning the merged path if there is majoirty subcontext available
+      return subcontextData.isSubcontext ? { path: newPath.slice(0, subcontextData.index + 1), lastUpdated: timestamp() } : recentlyEditedThought
```

Comment:
If there is a `recentlyEditedThought` with a majority subcontext, it should be moved to the top of the recently edited list, not replaced inline.

Reply:
Since we are merging then, we can just replace with the merged path and update it's timestamp. It will show up to the top of the recently edited list because of the updated timestamp.

Reply:
Good point. If that's the case though, why sort before merging? It seems like it should sort at the end.

Reply:
Yes that's a good point. We can just sort at the end.